shardingsphere-agent/shardingsphere-agent-core/src/main/java/org/apache/shardingsphere/agent/core/plugin/Loader.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.agent.core.plugin;
+
+import net.bytebuddy.description.type.TypeDescription;
+import org.apache.shardingsphere.agent.api.point.PluginInterceptorPoint;
+
+public interface Loader {
+    /**
+     * To detect the type whether or not exists.
+     *
+     * @param typeDescription TypeDescription
+     * @return contains when it is true
+     */
+    boolean containsType(TypeDescription typeDescription);
+    
+    /**
+     * Load plugin interceptor point by TypeDescription.
+     *
+     * @param typeDescription TypeDescription
+     * @return plugin interceptor point
+     */
+    PluginInterceptorPoint loadPluginInterceptorPoint(TypeDescription typeDescription);
+    
+    /**
+     * To get or create instance of the advice class. Create new one and caching when it is not exist.
+     *
+     * @param classNameOfAdvice class name of advice
+     * @param <T> advice type
+     * @return instance of advice
+     */
+    <T> T getOrCreateInstance(String classNameOfAdvice);
```

Comment:
It is better to rename `classNameOfAdvice` to `adviceClassName`

Reply:
it's better.